File: FlexiStore/src/app/shared/product.ts
export interface MainCategory {
	id: number;
	name: string;
	description: string;
}

export interface Brand {
	id: number;
	name: string;
}

export interface IProduct {
	id: number;
	mainCategory: MainCategory;
	size: string;
	price: number;
	title: string;
	description: string;
	brand: Brand;
}

File: FlexiStore/Data/Entities/Product.cs
namespace FlexiStore.Data.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public Category MainCategory { get; set; }
        public string Size { get; set; }
        public decimal Price { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public Brand Brand { get; set; }
    }
}